lib/ain-rs-exports/src/evm.rs

```diff
@@ -438,75 +437,61 @@ pub fn evm_try_get_block_number_by_hash(
 
 pub fn evm_create_dst20(
     result: &mut ffi::CrossBoundaryResult,
-    context: u64,
+    queue_id: u64,
     native_hash: [u8; 32],
     name: &str,
     symbol: &str,
     token_id: &str,
 ) {
-    match create_dst20(context, native_hash, name, symbol, token_id) {
-        Ok(_) => cross_boundary_success(result),
+    let address = ain_contracts::dst20_address_from_token_id(token_id);
+    debug!("Deploying to address {:#?}", address);
+    match address {
+        Ok(address) => {
+            let system_tx = QueueTx::SystemTx(SystemTx::DeployContract(DeployContractData {
+                name: String::from(name),
+                symbol: String::from(symbol),
+                address,
+            }));
+
+            match SERVICES
+                .evm
+                .queue_tx(queue_id, system_tx, native_hash, U256::zero())
+            {
+                Ok(_) => cross_boundary_success(result),
+                Err(e) => cross_boundary_error_return(result, e.to_string()),
+            }
+        }
```

Comment:
This is more readable IMO.

```suggestion
    let address = match ain_contracts::dst20_address_from_token_id(token_id) {
        Ok(address) => address,
        Err(e) => cross_boundary_error_return(result, e.to_string()),
    };

    debug!("Deploying to address {:#?}", address);

    let system_tx = QueueTx::SystemTx(SystemTx::DeployContract(DeployContractData {
        name: String::from(name),
        symbol: String::from(symbol),
        address,
    }));

    match SERVICES
        .evm
        .queue_tx(queue_id, system_tx, native_hash, U256::zero())
    {
        Ok(_) => cross_boundary_success(result),
        Err(e) => cross_boundary_error_return(result, e.to_string()),
    }
```

Reply:
Resolved in commit [590340a](https://github.com/DeFiCh/ain/pull/2277/commits/590340a4022596bb9f0d9dfbfa8ed49502290d01)